src/modules/layout/components/Header.js

```diff
@@ -61,11 +61,35 @@ type PropsType = {|
   scenes: Array<NavigationScene>,
   t: TFunction,
   theme: ThemeType,
+  peeking: boolean | 'unsure',
   navigateToLanding: () => void,
   routeKey: string
 |}
 
-class Header extends React.PureComponent<PropsType> {
+type StateType = {|
+  peeking: boolean
+|}
+
+class Header extends React.PureComponent<PropsType, StateType> {
+  constructor () {
+    super()
+
+    this.state = {
+      peeking: true
+    }
+  }
+
+  static getDerivedStateFromProps (props: PropsType, state: StateType): StateType {
```

Comment:
It should be pretty easy to get rid of this function. Since it's not necessary and `getDerivedStateFromProps` should be used rarely, we should avoid it here.

Reply:
Provide me with an other way of doing this :) The input of the props is "cached" in the state. There is only this way to do it

Reply:
Well the exact same behavior is not possible (but actually it is): Once we know that the route is peeking or not this property doesn't change anymore (especially it's never going to be set to `unsure` anyhow). You could just make `peeking` a nullable prop that defaults to `true` (or decide in the container to use `true` if it's unsure). Then you would get the same behavior as far as I can see...

Reply:
I'll look over it but not sure whether you are right.